### PR TITLE
Prot Warrior T28 Profile/APL

### PR DESCRIPTION
**Pull request by Mwahi**
Nomerature's T28 Profile/APL update
https://www.raidbots.com/simbot/report/8wSZcYWp6XiiV9fd3VU5Yt

### CODE REVIEW

File: engine/class_modules/sc_warrior.cpp [hunk 1]
```diff
@@ -7092,35 +7092,35 @@ void warrior_t::apl_prot()
...
   default_list -> add_action( "run_action_list,name=aoe,if=spell_targets.thunder_clap>=3" );
   default_list -> add_action( "call_action_list,name=generic" );
 
   generic -> add_talent( this, "Ravager" );
   generic -> add_talent( this, "Dragon Roar" );
-  generic -> add_action( this, "Shield Slam", "if=buff.shield_block.up" );
-  generic -> add_action( this, "Thunder Clap", "if=(spell_targets.thunder_clap>1|cooldown.shield_slam.remains)&talent.unstoppable_force.enabled&buff.avatar.up" );
-  generic -> add_action( this, "Shield Slam" );
-  generic -> add_action( this, covenant.condemn, "condemn");
+  generic -> add_action( this, "Shield Slam", "if=buff.shield_block.up||buff.outburst.up&rage<=55" );
```
Comment (Armadk) [inline engine/class_modules/sc_warrior.cpp]:
```suggestion
  generic -> add_action( this, "Shield Slam", "if=buff.shield_block.up|buff.outburst.up&rage<=55" );
```

File: engine/class_modules/sc_warrior.cpp [hunk 2]
```diff
@@ -7092,35 +7092,35 @@ void warrior_t::apl_prot()
 
   default_list -> add_action( "auto_attack" );
   default_list -> add_action( this, "Charge", "if=time=0" );
-  default_list -> add_action( "heroic_charge", "if=runeforge.reprisal" );
+  default_list -> add_action( this, "heroic_charge,if=rage<=40");
```
Comment (nyterage) [inline engine/class_modules/sc_warrior.cpp]: ``
```suggestion
  default_list -> add_action( "heroic_charge,if=rage<=40" );
```

or 

```suggestion
  default_list -> add_action( this, "Heroic Charge", "if=rage<=40" );
```

current line is causing it to error out looking for an ability called "heroic_charge,if=rage<=40"

**Review by nyterage — COMMENTED** [review]
putting through one of these changes should fix the CI error

File: engine/class_modules/sc_warrior.cpp [hunk 3]
```diff
@@ -7092,35 +7092,35 @@ void warrior_t::apl_prot()
...
-  default_list -> add_action( "conquerors_banner", "if=runeforge.glory");
-  default_list -> add_action( this, "Demoralizing Shout", "if=talent.booming_voice.enabled" );
   default_list -> add_action( this, "Avatar" );
+  default_list -> add_action( "potion,if=buff.avatar.up|target.time_to_die<25" );
+  default_list -> add_action( "conquerors_banner", "if=buff.conquerors_banner.down");
   default_list -> add_action( "ancient_aftershock");
   default_list -> add_action( "spear_of_bastion");
-  default_list -> add_action( "conquerors_banner");
-  default_list -> add_action( this, "Shield Block", "if=buff.shield_block.down" );
+  default_list -> add_action( this, "Ignore Pain", "if=target.health.pct>=20&(target.health.pct>=80&!covenant.venthyr)&(rage>=85&cooldown.shield_slam.ready|rage>=60&cooldown.demoralizing_shout.ready&talent.booming_voice.enabled|rage>=70&cooldown.avatar.ready|rage>=40&cooldown.demoralizing_shout.ready&talent.booming_voice.enabled&buff.last_stand.up|rage>=55&cooldown.avatar.ready&buff.last_stand.up|rage>=80|rage>=55&cooldown.shield_slam.ready&buff.outburst.up||rage>=30&cooldown.shield_slam.ready&buff.outburst.up&buff.last_stand.up),use_off_gcd=1");
```
Comment (nyterage) [inline engine/class_modules/sc_warrior.cpp]:
```suggestion
  default_list -> add_action( this, "Ignore Pain", "if=target.health.pct>=20&(target.health.pct>=80&!covenant.venthyr)&(rage>=85&cooldown.shield_slam.ready|rage>=60&cooldown.demoralizing_shout.ready&talent.booming_voice.enabled|rage>=70&cooldown.avatar.ready|rage>=40&cooldown.demoralizing_shout.ready&talent.booming_voice.enabled&buff.last_stand.up|rage>=55&cooldown.avatar.ready&buff.last_stand.up|rage>=80|rage>=55&cooldown.shield_slam.ready&buff.outburst.up|rage>=30&cooldown.shield_slam.ready&buff.outburst.up&buff.last_stand.up),use_off_gcd=1");
```

**Review by nyterage — DISMISSED** [review]

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 4]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
-covenant=night_fae
-soulbind=grove_invigoration/niyas_tools_burrs/bonded_hearts/ashen_juggernaut:11:1/unnerving_focus:11:1/destructive_reverberations:11:1
+talents=3122011
+covenant=necrolord
+soulbind=volatile_solvent/oozs_frictionless_coating/Kevins_oozeling/ashen_juggernaut:11:1/unnerving_focus:11:1/veterans_repute:11:1/adaptive_armor_fragment:11:1
 renown=80
 
-head=gaze_of_the_infinite_infantry,id=188942,bonus_id=7187/1498
-neck=worldkiller_iris,id=189859,bonus_id=7187/1524
+head=shadowghast_helm,id=171415,bonus_id=6647/6649/6716/8130/1588,gem_id=173128
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
head=shadowghast_helm,id=171415,bonus_id=6647/6649/6716/8130/1588/6935,gem_id=173128
```

6935 for items with a socket

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 5]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
+neck=cabochon_of_the_infinite_flight,id=185820,bonus_id=3536/4786/1808/1595
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
neck=cabochon_of_the_infinite_flight,id=185820,bonus_id=1595/6536/6646/6935,gem_id=173128
```
allowed to use a socket

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 6]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
 shoulders=pauldrons_of_the_infinite_infantry,id=188941,bonus_id=1505/7187
 back=shroud_of_the_sires_chosen,id=189847,bonus_id=7187/1524
 chest=breastplate_of_the_infinite_infantry,id=188938,bonus_id=1505/7187,enchant=eternal_stats
-wrists=veiled_auctioneers_vambraces,id=189825,bonus_id=7187/1524
-hands=handguards_of_hidden_realms,id=189850,bonus_id=7187/1524,enchant=eternal_strength
-waist=shadowghast_waistguard,id=171418,bonus_id=6648/6649/7451/8130/1588,gem_id=173128
+wrists=vambraces_of_the_infinite_infantry,id=188943,bonus_id=1505/7187,gem_id=173128
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
wrists=vambraces_of_the_infinite_infantry,id=188943,bonus_id=1505/7187/6935,gem_id=173128
```
6935 bonus id

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 7]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
+hands=grasps_of_the_infinite_infantry,id=188937,bonus_id=1505/7187,enchant=eternal_strength
+waist=girdle_of_the_soulflayer,id=179326,bonus_id=1592/7187
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
waist=girdle_of_the_soulflayer,id=179326,bonus_id=1592/6536/6646/6935,gem_id=173128
```
waist is allowed a gem

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 8]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
 legs=legplates_of_the_infinite_infantry,id=188940,bonus_id=7187/1498
-feet=shadowghast_sabatons,id=171413,bonus_id=6648/6649/7451/6969/1588,gem_id=173128
-finger1=loquacious_keepers_peridot,id=189802,bonus_id=7187/1524,enchant=tenet_of_haste
-finger2=rygelons_heraldric_ring,id=189854,bonus_id=7187/1524,enchant=tenet_of_haste
-trinket1=scars_of_fraternal_strife,id=188253,bonus_id=7187/1524
-trinket2=old_warriors_soul,id=186438,bonus_id=7187/1498
+feet=shadowghast_sabatons,id=171413,bonus_id=6647/6649/6716/6969/1588
+finger1=bloodoath_signet,id=178871,bonus_id=3536/4786/1808/1592,gem_id=173128,enchant=tenet_of_haste
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
finger1=bloodoath_signet,id=178871,bonus_id=1592/6536/6646/6935,gem_id=173128,enchant=tenet_of_haste
```
socket, m+ id's

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 9]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
+finger2=arachnid_cipher_ring,id=178933,bonus_id=1592/7187,gem_id=173128,enchant=tenet_of_haste
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
finger2=arachnid_cipher_ring,id=178933,bonus_id=1592/6536/6646/6935,gem_id=173128,enchant=tenet_of_haste
```
socket + m+ bonusid

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 10]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
+trinket1=the_first_sigil,id=188271,bonus_id=1524
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
trinket1=the_first_sigil,id=188271,bonus_id=1524/7187
```

File: profiles/generators/Tier28/T28_Generate_Warrior.simc [hunk 11]
```diff
@@ -128,25 +128,25 @@ race=draenei
...
+trinket2=ticking_sack_of_terror,id=185844,bonus_id=1595/7187
```
Comment (Armadk) [inline profiles/generators/Tier28/T28_Generate_Warrior.simc]:
```suggestion
trinket2=ticking_sack_of_terror,id=185844,bonus_id=1595/6536/6646
```

**Review by Armadk — CHANGES_REQUESTED** [review]
I added a few gems in the sockets where they were missing, and updated all the bonus id's so they should be correct now.  Used all haste gems, in keeping with what had already been used.

**Review by Armadk — APPROVED** [review]

**Pull request merged**